R/vb_epidist.R

```diff
@@ -0,0 +1,309 @@
+#' Constructor for `vb_epidist` class
+#'
+#' @description Create an `vb_epidist` object by binding two `epidist` objects
+#' and assigning the `vb_epidist` class. Not extra checks are made as these
+#' take place in the `epidist` helper function.
+#'
+#' @inheritParams vb_epidist
+#'
+#' @return `vb_epidist` object
+#' @keywords internal
+new_vb_epidist <- function(intrinsic_epidist,
+                           extrinsic_epidist) {
+
+  # return vb_epidist object
+  structure(
+    list(
+      intrinsic = intrinsic_epidist,
+      extrinsic = extrinsic_epidist),
+    class = "vb_epidist"
+  )
+}
+
+#' Create an `vb_epidist` object
+#'
+#' @param intrinsic_epidist An `epidist` object
+#' @param extrinsic_epidist An `epidist` object
+#'
+#' @description The `vb_epidist` class is an extension of the `epidist` class
+#' (although not a subclass of `epidist`). It is is used to store
+#' epidemiological parameters for vector-borne diseases. It has the same
+#' methods (`print()`, `format()`, `plot()`, `generate()`, `cdf()`, `density()`,
+#' `quantile()`) as the `epidist` class and therefore there are used
+#' identically.
+#'
+#' @details The `epidist` objects should contain metadata (`epidist$metadata`)
+#' indicating it is a vector-borne disease
+#' (`epidist$metadata$vector_borne` is TRUE) and the extrinsic distribution
+#' should indicate in the metadata that it is the extrinsic distribution
+#' (`epidist$metadata$extrinsic` is TRUE). If these two aspects are not given
+#' the construction of the class will throw a warning.
+#'
+#' @seealso [`epidist()`]
+#'
+#' @return `vb_epidist` object
+#' @export
+#'
+#' @examples
+#' vb <- vb_epidist(
+#'   intrinsic_epidist = epidist(
+#'     disease = "dengue",
+#'     pathogen = "dengue_virus",
+#'     epi_dist = "incubation_period",
+#'     prob_distribution = "gamma",
+#'     prob_distribution_params = c(shape = 1, scale = 1),
+#'     metadata = create_epidist_metadata(
+#'       vector_borne = TRUE,
+#'       extrinsic = FALSE
+#'     )
+#'   ),
+#'   extrinsic_epidist =  epidist(
+#'     disease = "dengue",
+#'     pathogen = "dengue_virus",
+#'     epi_dist = "incubation_period",
+#'     prob_distribution = "gamma",
+#'     prob_distribution_params = c(shape = 2, scale = 2),
+#'     metadata = create_epidist_metadata(
+#'       vector_borne = TRUE,
+#'       extrinsic = TRUE
+#'     )
+#'   )
+#' )
+vb_epidist <- function(intrinsic_epidist,
+                       extrinsic_epidist) {
+
+  # check input
+  validate_epidist(intrinsic_epidist)
+  validate_epidist(extrinsic_epidist)
+
+  # call vb_epidist constructor
+  vb_epidist <- new_vb_epidist(
+    intrinsic_epidist = intrinsic_epidist,
+    extrinsic_epidist = extrinsic_epidist
+  )
+
+  # call epidist validator
+  validate_vb_epidist(vb_epidist)
+
+  # return vb_epidist object
+  vb_epidist
+}
```

Comment:
1. Typo in function title.
2. Expand on what the `intrinsic_epidist` and `extrinsic_epidist` objects refer to, and especially which one refers to the vector. Suggest renaming these to "human_epidist" and "vector_epidist" for better understanding.
3. Plot and print methods: See feedback on `epidist` objects, same applies here. Suggest using `{ggplot2}` to avoid difficult placement of plot titles. Adds a dependency but possibly useful in the long run, especially if seeking to also add uncertainty visualisations. Not a massive priority.

Reply:
I'm not going to change the naming of the `vb_epidist` slots at the moment, but have opened an issue (#93) in order to reach a consensus on which naming is better. I'm happy to change just want to make sure it is an improvement beforehand.

Reply:
Resolved in 300615f84572cc791c2c006cdee61fc2a959f475.

Reply:
Thanks both fair. I've no inherent problem with the naming.